src/main/java/com/google/cloud/tools/gradle/appengine/util/GradleCompatibility.java

```diff
@@ -0,0 +1,51 @@
+package com.google.cloud.tools.gradle.appengine.util;
+
+import java.io.File;
+import org.gradle.api.tasks.bundling.AbstractArchiveTask;
+import org.gradle.util.GradleVersion;
+
+/** Utility class for Gradle compatibility related functions. As an end user, do not use. */
+public class GradleCompatibility {
+
+  /**
+   * Compatibility method for getting the archive location.
+   *
+   * <p>Illustrious history of why deprecations should be handled as soon as they come up:
```

Comment:
Thank you for explaining the detailed context and history! Could we move them to be documented in the PR description for this contribution, instead of inside the javadoc comments here?

Reply:
Anyone who maintains the code will need that information ready and easily accessible. In the future Gradle very likely will change things again and this will need to be updated. Or when the minimum Gradle version of GAE plugin increases, you'll also need this to be able to confidently remove branches. I strongly believe it should stay there, maybe with slightly less snark ;)

Note: I've done the same for my own plugins and these comments helped a lot when I got an error or warning near these functions. Formatting and location of the file can change but the contents usually follow that, and therefore it'll be readily available where and when needed. Very few will find the PR, actually someone might only do that if there's 0 docs. Even then it'll more assumption than investigation.

I'm happy to move it away from the **public** Javadoc into the **inside** of the function though. It's just not possible to use formatting there, but markdown would still be readable.

What do you think?

Reply:
Agree with you on the benefit for maintainability, though I think the key pieces of information in this timeline that someone would look for, in the scenarios you've described and given the current logic for this utility method, are the earliest gradle versions with `getArchiveFile` added (5.1.0-M1), and with `getArchivePath` removed  (likely Gradle 9).    

The intermediate deprecation/nagging history is super helpful context, but a bit too verbose for public javadoc. If you could help move them to the inside of the function, that would be great (an alternative to keep formatting could also be moving it to this PR description, and adding an in-line comment that links to the PR)

Reply:
https://github.com/GoogleCloudPlatform/app-gradle-plugin/pull/451/commits/0429c39024660215031a21affece88f4a1d8d1f2